File: source/algorithm/NQ006/NQ006_improved.py
# NQ006 成绩分区
import sys
def main():
    for line in sys.stdin:
        line=line.strip()
        if not line:continue
        s=float(line)
        if s<0 or s>100:print("Wrong Score!")
        elif s>=90:print("A:[90,100]")  # A级
        elif s>=80:print("B:[80,90)")   # B级
        elif s>=70:print("C:[70,80)")   # C级
        elif s>=60:print("D:[60,70)")   # D级
        else:print("E:[0,60)")         # E级
if __name__=="__main__":main()

File: source/algorithm/NQ006/NQ006_improved.cpp
// NQ006 成绩分区
#include<iostream>
using namespace std;
int main(){
    double s;
    while(cin>>s){  // 循环读取输入
        if(s<0||s>100)cout<<"Wrong Score!\n";
        else if(s>=90)cout<<"A:[90,100]\n";  // A级
        else if(s>=80)cout<<"B:[80,90)\n";   // B级
        else if(s>=70)cout<<"C:[70,80)\n";   // C级
        else if(s>=60)cout<<"D:[60,70)\n";   // D级
        else cout<<"E:[0,60)\n";             // E级
    }
    return 0;
}